changelog/upgrade-notes.md

```diff
@@ -2,6 +2,20 @@
 
 ### Next Release
 
+#### Behavior of `minZoom` and `maxZoom` on layers
+
+The `minZoom` on `ol/layer/Base` and all subclasses is now inclusive, `maxZoom` is exclusive. This is the opposite of the previous behavior. This should be the desired and more intuitive behavior, so workarounds like subtracting a very small number from the desired `minZoom` like
+```js
+new Layer({
+  minZoom: desiredMinZoom - 1e-12
+})
+```
```

Comment:
Is there a ticket or other question online where you have seen people do this?

Reply:
Not directly. The Mapbox/MapLibre style spec uses inclusive minzoom and exclusive maxzoom: https://docs.mapbox.com/style-spec/reference/layers/#minzoom. The workaround itself was used in ol-mapbox-style in combination with the layer's maxResolution only, because back then the minZoom was not available yet: https://github.com/openlayers/ol-mapbox-style/blob/d4617d9fee7c294caba0044cd7423eb5b2e87f20/src/apply.js#L324.